BredaUniversityGames/Y2024-25-PR-BB
# Feature/steam build

### Description

- Added packaging files for distribution builds to steam
- Added `package_and_upload_steam_dev.bat`, which packages and uploads build to steam on our development steam branch
- Added a `Node` constructor as I got errors on trying to `emplace_back` the node struct without a matching constructor
- Added the `game` folder with wren scripts to the packaged build
- Edited the cmake executable name, as the Linux executable was named `Game` and the folder with wren scripts was named `game`, which blocked the build from being correctly packaged.

### Issues

[Codecks card](https://bubonic-brotherhood.codecks.io/decks/31-steam-build/card/1hk-steam-build-depot-for-development)

### Test criteria

Someone with upload permissions could test using the `package_and_upload_steam_dev.bat` script by:
1. Making sure you have built linux and windows release configurations
2. Running the `package_and_upload_steam_dev.bat` script
3. Filling in the log in information for Steam
4. Waiting for the upload to complete
5. Going to steamworks website and under `SteamPipe` -> `Builds` you should be able to see the uploaded build

To run the build itself:
1. Open Steam
2. Go to properties of the custom tech project
3. In the `Betas` tab fill in the branch password visible in the same view on Steamworks website `SteamPipe` -> `Builds` or view the password in my show-and-tell post about Steam Builds (IMPORTANT -> Don't share this password outside with the public!)
4. You should see beside the `Download` or `Play` button on Steam that you are on the `development` version of our game
5. (Download and) click play
